electrum/lnpeer.py

```diff
@@ -1484,3 +1426,53 @@ def verify_signature(tx, sig):
         # broadcast
         await self.network.try_broadcasting(closing_tx, 'closing')
         return closing_tx.txid()
+
+    @log_exceptions
+    async def htlc_switch(self):
+        while True:
+            await asyncio.sleep(0.1)
+            for chan_id, chan in self.channels.items():
+                if not chan.can_send_ctx_updates():
+                    continue
+                self.maybe_send_commitment(chan)
+                done = set()
+                unfulfilled = chan.hm.log.get('unfulfilled_htlcs', {})
```

Comment:
I think this kind of deep access into `chan.hm.log` should be abstracted away. We should add methods to `HTLCManager` for these specific cases if they don't yet exist.